modules/apps/portal-template/portal-template-soy-impl/src/main/java/com/liferay/portal/template/soy/internal/util/SoyTemplateResourcesCollector.java

```diff
@@ -156,19 +155,7 @@ protected String getTemplateId(long bundleId, URL url) {
 	private TemplateResource _getTemplateResource(String templateId, URL url)
 		throws TemplateException {
 
-		TemplateResource templateResource;
-
-		if (TemplateResourceLoaderUtil.hasTemplateResourceLoader(
-				TemplateConstants.LANG_TYPE_SOY)) {
-
-			templateResource = TemplateResourceLoaderUtil.getTemplateResource(
-				TemplateConstants.LANG_TYPE_SOY, templateId);
-		}
-		else {
-			templateResource = new URLTemplateResource(templateId, url);
-		}
-
-		return templateResource;
+		return new URLTemplateResource(templateId, url);
```

Comment:
Hi @dantewang,
Before, we should go into the first branch which is using TemplateResourceLoaderUtil to generate the URLTemplateResource.
After, you create URLTemplateResource directly.

The result should be same, but the difference is TemplateResourceLoaderUtil has a cache level, I think this is going to affect performance.

I searched for the usages in our code base, and found a lot of usages like this, that is to create a URLTemplateResource directly instead of using TemplateResourceLoaderUtil. This should be wrong, but we can not stop people doing so since URLTemplateResource is in kernel package.

An option to avoid URLTemplateResource created directly is to replace all the usages to use TemplateResourceLoaderUtil, and copy it to internal package and deprecate the kernel one.

The problem with that option is currently TemplateResourceLoaderUtil does not export easy API for people to use. Like in this example, if we use TemplateResourceLoaderUtil, we have to manipulate the template id and parse it to create URL, just like what we did in LPS-88867.

Could you please check all the usages of creating template resource directly, and see if we can create API for those usages, and then remove all the direct creating and deprecate the kernel class.

We should not do the removing like you did in this pull, so I will close it.

Thanks.
Tina.